Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -392,48 +390,64 @@ private extension CardReaderConfigError {
 
 // MARK: Networking Methods
 private extension CardPresentPaymentStore {
-    /// Switch the store to use WCPay as the backend.
-    /// Does nothing if the store is already using WCPay.
-    /// WCPay is the initial default for the store.
+    /// Sets the store to use a given payment gateway
     ///
-    func useWCPayAsBackend() {
-        guard usingBackend != .wcpay else {
-            return
-        }
-
-        usingBackend = .wcpay
-    }
-
-    /// Switch the store to use Stripe as the backend.
-    /// Does nothing if the store is already using Stripe.
-    ///
-    func useStripeAsBackend() {
+    func use(paymentGatewayAccount: PaymentGatewayAccount) {
         guard allowStripeIPP else {
             DDLogError("useStripeAsBackend called when stripeExtensionInPersonPayments disabled")
             return
         }
 
-        guard usingBackend != .stripe else {
+        guard paymentGatewayAccount.isWCPay else {
+            usingBackend = .stripe
             return
         }
 
-        usingBackend = .stripe
+        usingBackend = .wcpay
     }
 
     /// Loads the account corresponding to the currently selected backend. Deletes the other (if it exists).
     ///
     func loadAccounts(siteID: Int64, onCompletion: @escaping (Result<Void, Error>) -> Void) {
-        switch usingBackend {
-        case .wcpay:
-            loadWCPayAccount(siteID: siteID, onCompletion: onCompletion)
-        case .stripe:
-            loadStripeAccount(siteID: siteID, onCompletion: onCompletion)
+        var error: Error? = nil
+
+        let group = DispatchGroup()
+        group.enter()
+        loadWCPayAccount(siteID: siteID, onCompletion: { result in
```

Comment:
So, if we do it this way, we will always get this in the logs:

```2022-01-18 11:50:50.786968-0800 WooCommerce[5474:3383639] ⛔️ Error synchronizing WCPay Account: Dotcom Invalid REST Route```

for Stripe powered stores. Could we note use `usingBackend` to avoid this?

Reply:
No, we need to load both accounts, if we want the rest of the system to work correctly. I assume you see that error on stores that have only the Stripe account installed, but not WCPay, correct?